File: src/pages/farmPage/farmPageOrder/OrderDay.js
import React from "react";
import Order from "./Order";

const OrderDay = (props) => {


    const formatDate = date => {
        let newDate = new Date(date)
        return newDate.toLocaleDateString();
    }

    return <div className="orderDay__main">
        <h1>{formatDate(props.data.date)}</h1>
        <div className="orderDay__orders">
            {props.data.orders.map(order => {
                    return <Order key={order.id} data={order}  />
                }
            )}
        </div>
    </div>
}

export default OrderDay;

File: backend/src/main/java/com/de/boederij/payload/OrderDay.java
package com.de.boederij.payload;

import com.de.boederij.model.Order;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.Date;
import java.util.List;

@Data
@AllArgsConstructor
public class OrderDay {

    private Date date;
    private List<Order> orders;
}
